src/subscription.ts

```diff
@@ -88,6 +89,30 @@ export type SetSubscriptionMetadataResponse = MetadataResponse;
 export type DetachSubscriptionCallback = EmptyCallback;
 export type DetachSubscriptionResponse = EmptyResponse;
 
+// JSDoc won't see these, so this is just to let you get typings
+// in your editor of choice. Callback types are open ended so that
+// not-quite-matching parameters won't cause a breaking change. We
+// may still shift these to required later.
+export declare interface Subscription {
+  on(
+    event: 'message',
+    listener: (() => void) | ((message: Message) => void)
+  ): this;
+  on(
+    event: 'error',
+    listener: (() => void) | ((error: StatusError) => void)
+  ): this;
+  on(event: 'close', listener: () => void): this;
+
+  // Only used internally.
+  on(event: 'newListener', listener: Function): this;
+  on(event: 'removeListener', listener: Function): this;
+
+  // Catch-all. If you get an error about this line, it means you're
+  // using an unsupported event type or listener type.
+  on(event: string, listener: void): this;
```

Comment:
I don't fully understand why this is needed.  Are you saying that without it, using a non-enumerated event name doesn't cause a compile time error?

Reply:
TS produces unhelpful error messages otherwise, if you use a non-enumerated event, or use a bad type for the callback on an enumerated one. It'll say that your call to `on()` doesn't match the definition for whatever `on('foo')` is declared last.